client/core/core.go

```diff
@@ -4867,40 +4930,47 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 		return nil, newError(suspendedAcctErr, "may not trade while account is suspended")
 	}
 
-	corder, updatedAssets, err := c.prepareTrackedTrade(dc, form, crypter)
+	resultChan, err := c.prepareTrackedTrade(dc, form, crypter, queue)
 	if err != nil {
 		return nil, err
 	}
 
-	for assetID := range updatedAssets {
-		c.updateAssetBalance(assetID)
+	submissionResult := <-resultChan
```

Comment:
This pattern *looks* fully-synchronous because it appears to wait for an "order submission result" regardless of `queue`, but really if `queue=true`, it's not an order submission result - the submission server request is still in process.

In other words, `prepareTrackedTrade` might as well just always return (`core.Order`, `error).  The channel appears pointless, misleading at best.

`prepareTrackedTrade` should just be split in two pieces so that when `queue` is `true`, it launches part 2 in a goroutine, and when `false`, it runs part 2 blocking.

Reply:
>  /api/trade to return before the order submission request is fully handled.

I had tested about 10 sec delay after `err = dc.signAndRequest(msgOrder, route, result, fundingTxWait+DefaultResponseTimeout)` but not more than 10sec unless the order don't show up on orderbook.

`/api/trade` returns with temp order and the frontend get updated subsequently via note.